libmvsim/src/Sensors/LaserScanner.cpp

```diff
@@ -245,8 +244,8 @@ void LaserScanner::simul_post_timestep(const TSimulContext &context)
 		{
 			if (it->validRange[i])
 			{
-				lastScan->scan[i] = std::min(lastScan->scan[i], it->scan[i]);
-				lastScan->validRange[i] = 1; // valid
+				lastScan->setScanRange(i, std::min(lastScan->scan[i], it->scan[i]));
+				lastScan->setScanRangeValidity(i, true);; // valid
 			}
 		}
 	}
```

Comment:
That's all! 

You could try launching the demo:

`roslaunch mvsim mvsim_demo_2robots.launch `

to see if LiDAR scans look as expected...

Let me know if you need anything else... and thanks again for taking care of this update!